swap_meet/vendor.py

```diff
@@ -1,2 +1,79 @@
+from .item import Item
+
+# wave 1
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory = None):
+        self.inventory = [] if inventory is None else inventory
+    
+    def add(self,item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self,item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+    
+#wave 2
+    def get_by_id(self, id):
+        for item in self.inventory:
+            if item.id == id:
+                return item
+        else:
+            return None
+
+
+#wave 3
+    def swap_items(self,other_vendor, my_item, their_item):
+        if ((my_item not in self.inventory) or
+            (their_item not in other_vendor.inventory)):
+            return False
+        self.inventory.remove(my_item)
```

Comment:
I'd probably add a blank line between lines 31 and 32 to improve readability by logically and visually separating different parts of the code
```suggestion

        self.inventory.remove(my_item)
```